doc/source/user/basics.broadcasting.rst

```diff
@@ -70,14 +70,18 @@ It starts with the trailing (i.e. rightmost) dimensions and works its
 way left.  Two dimensions are compatible when
 
 1) they are equal, or
-2) one of them is 1
+2) one of them is 1.
 
 If these conditions are not met, a
 ``ValueError: operands could not be broadcast together`` exception is
-thrown, indicating that the arrays have incompatible shapes. The size of
-the resulting array is the size that is not 1 along each axis of the inputs.
+thrown, indicating that the arrays have incompatible shapes.
 
-Arrays do not need to have the same *number* of dimensions.  For example,
+Arrays do not need to have the same *number* of dimensions.  Resulting
```

Comment:
Something like:
```
Arrays do not need to have the same *number* of dimensions.  The resulting
array will have the same number of dimensions as the input array with the
greatest number of dimensions where the size of each dimension is the largest
size of the corresponding dimension of the input arrays. Note that missing
dimensions are assumed to have size one
```

Reply:
Thanks Charles.

I changed text along the lines of your proposal.

I highlighted *size* to help some readers clearly see the difference between the number of dimensions and their size. Some of the people reading this will be new enough for this hint to be helpful, I think.

I kept sentence `Note that missing dimensions are assumed to have size one` that you proposed - to avoid another back-and-forth - in case you feel strongly about its inclusion. But I wonder if it adds more confusion than clarity. I think some people who are relatively new to numpy (who may be the majority of eyes on this particular page) may find it confusing: "what missing dimensions? missing from where?" I think without that sentence - even if the explanation is not 100% logically complete - things might be more clear. At the end of the day, missing dimensions may just be missing (and not of the assumed/imaginary size one) and all the logic about dimensions and their size still holds.

Reply:
Rebased to the tip of main now.